uint8 streamId
uint8[] data
bool verifyReceipt
